tree.rb

```diff
@@ -0,0 +1,123 @@
+# ! MINOR BUGS: 
+# 1) :text is being deleted from the <ul> node. I know this has to do with 
+# the special gsub for <li> children but not specifically why it's happening.
+# 2) Spacing error on some of the other :text fields. Specifically the 'outer-div'
+
+# Create a DOM Tree Class
+
+# First thing is to create the Node struct
+Node = Struct.new(:name, :text, :classes, :id, :children, :parent)
+
+class DOMTree
+	attr_reader :file
+	attr_accessor :head
+
+	def initialize(file_name)
+		# gsub(/<!.*?>/, "") Removes <! doctype > from the document
+		# gsub(/>\s*</,"><") Removes all the whitespace from between > & < characters
+		# gsub(/\n/, "") Removes the newline characters from the document
+		@file = load(file_name).read.gsub(/\n/, "").gsub(/>\s*</,"><").gsub(/<!.*?>/, "")
```

Comment:
Would make this a separate method, ideally using something like the dictionary reader modularity where the loading is handled by a separate service